### PR TITLE
[Markdown] Replace custom css styles and native html with OUI.

### PR DESCRIPTION
**Pull request by curq**
### Description

Replaces custom styles and native html with OUI for better OUI compliance. After this change only one custom style will remain.
```
.visEditor--markdown {
  .visEditorSidebar__config > * {
    flex-grow: 1;
  }
}
```
This style related to the `vis_default_editor` that is currently used by visualization plugins. Specifically, this part is targeted. 
https://github.com/opensearch-project/OpenSearch-Dashboards/blob/fe1f1d18ea9b26ddb19ae805f676b6e09798b294/src/plugins/vis_default_editor/public/_sidebar.scss#L29-L35

---

`.mkdEditor`, `.mkdVis` and `.visEditor--markdown__textarea` were removed.

### Issues Resolved

Solves https://github.com/opensearch-project/OpenSearch-Dashboards/issues/4389

## Screenshot
Targeted elements:

**.mkdEditor**
image

**.visEditor--markdown__textarea**
image

**.mkdVis**
image

## Testing the changes

The previous behavior should remain the same. To test compare main version and this PR version, especially, the targeted elements for which screenshot were provided above.

### Check List

- [ ] All tests pass
  - [x] `yarn test:jest`
  - [x] `yarn test:jest_integration`
  - [ ] `yarn test:ftr`
- [ ] New functionality includes testing.
- [ ] New functionality has been documented.
- [x] Update [CHANGELOG.md](./../CHANGELOG.md)
- [x] Commits are signed per the DCO using --signoff

### CODE REVIEW

**Comment by joshuarrrr** [comment]
@curq For completeness, can you also include screenshots of the markdown visualization as it appears when added to a dashboard?

**Review by joshuarrrr — COMMENTED** [review]
Code looks good - I'd like a little more time to check it out and play with it.

**Comment by curq** [comment]
> @curq For completeness, can you also include screenshots of the markdown visualization as it appears when added to a dashboard?

@joshuarrrr Here are before/after screenshot for markdown in dashboard view.

### Before
before
selectedbefore

### After
after

selectedAfter

Although the way container for markdown is rendered (now it is full height) is changed, I didn't notice any visual differences. The resizing of panel works the same.

**Comment by joshuarrrr** [comment]
All checks passed

**Review by joshuarrrr — APPROVED** [review]
Looks great!

File: CHANGELOG.md [hunk 1]
```diff
@@ -72,6 +72,7 @@ Inspired from [Keep a Changelog](https://keepachangelog.com/en/1.0.0/)
 - Migrate from legacy elasticsearch client to opensearch-js client in `osd-opensearch-archiver` package([#4142](https://github.com/opensearch-project/OpenSearch-Dashboards/pull/4142))
 - Replace the use of `bluebird` in `saved_objects` plugin ([#4026](https://github.com/opensearch-project/OpenSearch-Dashboards/pull/4026))
 - Relocate tutorials imagery into `src/plugins/home/public/assets/tutorials/logos` ([#4382](https://github.com/opensearch-project/OpenSearch-Dashboards/pull/4382))
+- [Markdown] Replace custom css styles and native html with OUI ([#4390](https://github.com/opensearch-project/OpenSearch-Dashboards/pull/4390))
```
Comment (AMoo-Miki) [inline CHANGELOG.md:75]:
```suggestion
- [Markdown] Replace custom CSS styles and HTML markup with OUI components ([#4390](https://github.com/opensearch-project/OpenSearch-Dashboards/pull/4390))
```

**Review by joshuarrrr — APPROVED** [review]

**Pull request merged**